compile_docker.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+
+BASE_IMAGE="ubuntu:jammy-20220421"
+
+IMAGE_NAME="scons_image"
+CONTAINER_NAME="scons_container"
+
+COMPILER="https://developer.arm.com/-/media/Files/downloads/gnu/11.2-2022.02/binrel/gcc-arm-11.2-2022.02-x86_64-arm-none-eabi.tar.xz"
+COMPILER_UNTAR=embedded/toolchains/gcc-arm-none-eabi/
+COMPILER_TAR=embedded/toolchains/gcc-arm-none-eabi.tar.xz
+
+# wrapper variables 
+COMPONENT=""
+
+# pull base image 
+docker pull ${BASE_IMAGE}
+	
+# build base ubuntu image  
+docker build -t ${IMAGE_NAME} .
+
+docker-compose up -d
+
+# download compiler 
+
+# check for  through repo dir for compiler 
+
+if [ -d ${COMPILER_UNTAR} ]; then
+
+	echo "Compiler files present. Skipping download"
+else
+	if [ ! -f ${COMPILER_TAR} ]; then         
+        
+		echo "No compiler files found. Downloading."
+		docker exec ${CONTAINER_NAME} wget ${COMPILER} -O embedded/toolchains/gcc-arm-none-eabi.tar.xz
+		
+		if [ $? -ne 0 ]; then
+			echo "Compiler download failed. Exiting"
+			exit 1
+		fi  
+	fi		
+	
+	#extract and move to appropriate dir 
+        echo "Extracting tar file..."
+	mkdir embedded/toolchains/gcc-arm-none-eabi
+	docker exec ${CONTAINER_NAME} tar -xf ${COMPILER_TAR} -C embedded/toolchains/gcc-arm-none-eabi/ --strip-components 1
+
+fi
+
+# open containers shell
+docker exec -it ${CONTAINER_NAME} bash
```

Comment:
add `--rm` to this command (after `-it`) so that you don't get a new container every time you run it